fcsparser/api.py

```diff
@@ -375,20 +375,26 @@ def read_text(self, file_handle):
         else:
             self.channel_numbers = range(1, pars + 1)  # Channel numbers start from 1
 
+        
         # Extract parameter names
-        try:
-            names_n = tuple([text["$P{0}N".format(i)] for i in self.channel_numbers])
-        except KeyError:
-            names_n = []
-
-        try:
-            names_s = tuple([text["$P{0}S".format(i)] for i in self.channel_numbers])
-        except KeyError:
-            names_s = []
-
-        self.channel_names_s = names_s
-        self.channel_names_n = names_n
-
+        channel_names_n = []
+        channel_names_s = []
+        for i in self.channel_numbers:
```

Comment:
sting interpolation using named arguments rather than indexed arguments

```python

for channel_number in self.channel_numbers:
   n_key = f'$P{channel_number}N'
   s_key = f'$P{channel_number}S'
```